tensorflow/c/experimental/filesystem/modular_filesystem_test.cc

```diff
@@ -85,17 +85,34 @@ class ModularFileSystemTest : public ::testing::TestWithParam<std::string> {
     const std::string test_name = tensorflow::str_util::StringReplace(
         ::testing::UnitTest::GetInstance()->current_test_info()->name(), "/",
         "_", /*replace_all=*/true);
+    // Since we need the tests for cloud filesystem to run on all OSs (Windows,
+    // MacOS, Linux, ...) The path to temp directory must not be dependent on
+    // the OS which runs the tests.
+    const std::string tmp_dir_ =
+        cloud_path_.empty() ? ::testing::TempDir() : "/tmp/";
     root_dir_ = tensorflow::io::JoinPath(
-        ::testing::TempDir(),
+        tmp_dir_,
         tensorflow::strings::StrCat("tf_fs_", rng_val_, "_", test_name));
+    if (!GetParam().empty()) {
+      if (!cloud_path_.empty()) {
+        root_dir_ = tensorflow::strings::StrCat(GetParam(), "://", cloud_path_,
+                                                root_dir_, "/");
+      } else {
+        root_dir_ = tensorflow::strings::StrCat(GetParam(), "://", root_dir_);
+      }
+    }
     env_ = Env::Default();
   }
 
   void SetUp() override {
-    if (mkdir(root_dir_.c_str(), 0755) != 0) {
-      int error_code = errno;
-      GTEST_SKIP() << "Cannot create working directory: "
-                   << tensorflow::IOError(root_dir_, error_code);
+    FileSystem* fs = nullptr;
+    Status s = env_->GetFileSystemForFile(root_dir_, &fs);
+    if (fs == nullptr || !s.ok())
+      GTEST_SKIP() << "No filesystem registered: " << s;
+
+    s = fs->CreateDir(root_dir_);
+    if (!s.ok()) {
+      GTEST_SKIP() << "Cannot create working directory: " << s;
```

Comment:
We'll have to think how will this affect testing on filesystems which don't support directories at all.

Reply:
We could make this a flag too. However, it seems that there won't be a filesystem which don't support directories at all.